src/uv_append.c

```diff
@@ -332,7 +332,10 @@ static int uvAppendMaybeStart(struct uv *uv)
     }
 
     /* If there's a blocking barrier in progress, and it's not waiting for this segment
-     * to be finalized, let's wait. */
+     * to be finalized, let's wait.
+     *
+     * FIXME shouldn't we wait even if segment->barrier == uv->barrier, if there are
+     * other open segments associated with the same barrier? */
```

Comment:
Will we ever clear `uv->barrier` if we wait here if `segment->barrier == uv->barrier`? (I'm asking because I'm not as deep into it as you are right now :-))